assets/src/edit-story/app/uploader/useUploader.js

```diff
@@ -56,14 +65,55 @@ function useUploader(refreshLibrary = true) {
     [maxUpload]
   );
 
+  const validErrorMessage = createInterpolateElement(
+    sprintf(
+      __('Please choose only <b>%s</b> to upload.', 'web-stories'),
+      allowedMimeTypes.join(', ')
+    ),
+    {
+      b: <b />,
+    }
+  );
+
+  const sizeErrorMessage = sprintf(
+    __(
+      'Your files is larger than the upload limit. The upload limit is %sMB. Please resize and try again!',
+      'web-stories'
+    ),
+    bytesToMB(maxUpload)
+  );
+
   const uploadFile = (file) => {
     // TODO Add permission check here, see Gutenberg's userCan function.
     if (!fileSizeCheck(file)) {
-      throw new Error('File size error');
+      const SizeError = new Error();
+      SizeError.name = 'SizeError';
+      SizeError.file = file.name;
+      SizeError.message = sprintf(
+        __(
+          'Your file is %sMB and the upload limit is %sMB. Please resize and try again!',
+          'web-stories'
+        ),
+        bytesToMB(file.size),
+        bytesToMB(maxUpload)
+      );
+      throw SizeError;
     }
 
     if (!isValidType(file)) {
-      throw new Error('File type error');
+      const ValidError = new Error();
+      ValidError.message = createInterpolateElement(
+        sprintf(
+          __('Please choose only <b>%s</b> to upload.', 'web-stories'),
```

Comment:
This needs a comment to explain the placeholder. See https://github.com/google/web-stories-wp/blob/9df43e73705c77b718d2fee9f5fed4ff779370b1/assets/src/edit-story/components/uploadDropTarget/message.js#L80

Reply:
Added